libs/portal-integration-angular/testing/primeng/p-button-directive.harness.ts

```diff
@@ -0,0 +1,35 @@
+import { BaseHarnessFilters, ComponentHarness, HarnessPredicate } from '@angular/cdk/testing'
+
+export interface PButtonDirectiveHarnessFilters extends BaseHarnessFilters {
+  id?: string
+}
+
+export class PButtonDirectiveHarness extends ComponentHarness {
```

Comment:
Have you tried to extend PButtonHarness and just overwrite the hostSelector?

Reply:
I did not think about that, however it seems like they do not have that much in common apart from id.

with() method should be overwritten to make sure that DirectiveHarness is being looked up (or cast from PButtonHarness to PButtonDirectiveHarness should be done in tests body)

click() method should be overwritten too

So extending only makes getId reusable. Also if any functionality would be added to PButtonHarness then we would need to check if the PButtonDirectiveHarness is not breaking.

Reply:
Then it is fine